src/lib.rs

```diff
@@ -359,19 +360,27 @@ where
     A: Annotation<KvPair<K, V>>,
     A::Archived: for<'any> CheckBytes<DefaultValidator<'any>>,
     I: Archive + for<'any> CheckBytes<DefaultValidator<'any>>,
+    <K as Archive>::Archived: Hash,
 {
     fn get(
         &self,
         key: &K,
     ) -> Option<MappedBranch<Self, A, I, MaybeArchived<V>>> {
-        self.walk(PathWalker::new(hash(key))).map(|branch| {
-            branch.map_leaf::<MaybeArchived<V>>(|kv| match kv {
-                MaybeArchived::Memory(kv) => MaybeArchived::Memory(kv.value()),
-                MaybeArchived::Archived(kv) => {
-                    MaybeArchived::Archived(kv.value())
-                }
+        self.walk(PathWalker::new(hash(key)))
+            .filter(|b| match b.leaf() {
+                MaybeArchived::Memory(kv) => hash(kv.key()) == hash(key),
```

Comment:
I'm pretty sure you can compare the keys directly, without hashing, here

Reply:
Problem was that K was not PartialEq, but I will add this constraint